### PR TITLE
Remove checking of miner funding from CreateNewBlock_validity test

### PR DESCRIPTION
**Pull request by schancel**
This commit stops CreateNewBlock_validity from checking miner funding
in the coinbase, as well as makes finding txIds in the blocks more
generic. The second change makes updating the test less fragile. The
specific ordering is no longer important since the introduction of
CTOR.

### CODE REVIEW

File: src/test/miner_tests.cpp [hunk 1]
```diff
@@ -776,6 +788,52 @@ BOOST_AUTO_TEST_CASE(CreateNewBlock_validity) {
...
+BOOST_AUTO_TEST_CASE(CreateNewBlock_minerfund) {
+    // Note that by default, these tests run with size accounting enabled.
+    GlobalConfig config;
+    // Don't check this functionality in this test.
+    config.SetEnableMinerFund(true);
+    const CChainParams &chainparams = config.GetChainParams();
+    CScript scriptPubKey =
+        CScript() << ParseHex("04678afdb0fe5548271967f1a67130b7105cd6a828e03909"
+                              "a67962e0ea1f61deb649f6bc3f4cef38c4f35504e51ec112"
+                              "de5c384df7ba0b8d578a4c702b6bf11d5f")
```
Comment (EyeOfPython) [inline src/test/miner_tests.cpp:801]: Can you use a different script? Uncompressed pubkey seems a bit unnecessary

File: src/test/miner_tests.cpp [hunk 2]
```diff
@@ -776,6 +788,52 @@ BOOST_AUTO_TEST_CASE(CreateNewBlock_validity) {
...
+                  << OP_CHECKSIG;
+    std::unique_ptr<CBlockTemplate> pblocktemplate;
+    CMutableTransaction tx;
+    CScript script;
+    TestMemPoolEntryHelper entry;
+    entry.nFee = 11 * SATOSHI;
+    entry.nHeight = 11;
```
Comment (EyeOfPython) [inline src/test/miner_tests.cpp:808]: Doesn't seem to be used? Please check if you can simplify this test.

File: src/test/miner_tests.cpp [hunk 3]
```diff
@@ -242,7 +247,14 @@ void MinerTestingSetup::TestPackageSelection(
     tx.vout[0].nValue = (100000000 - 10000) * SATOSHI;
     m_node.mempool->addUnchecked(entry.Fee(10000 * SATOSHI).FromTx(tx));
     pblocktemplate = AssemblerForTest(config).CreateNewBlock(scriptPubKey);
-    BOOST_CHECK_EQUAL(pblocktemplate->block.vtx[3]->GetId(), lowFeeTxId2);
+    const auto foundTx = std::find_if(
+        pblocktemplate->block.vtx.begin(), pblocktemplate->block.vtx.end(),
+        [lowFeeTxId2](const CTransactionRef &a) -> bool {
+            return a->GetId() == lowFeeTxId2;
+        });
```
Comment (EyeOfPython) [inline src/test/miner_tests.cpp:254]: Good. Thanks for cleaning this up finally

**Review by EyeOfPython — CHANGES_REQUESTED** [review]

File: src/test/miner_tests.cpp (hunk 2 above)
Comment (schancel) [reply src/test/miner_tests.cpp:808]: Yeah, I'll look through. Probably missed some copypasta.

File: src/test/miner_tests.cpp (hunk 3 above)
Comment (schancel) [reply src/test/miner_tests.cpp:254]: This was surprisingly painful to figure out. Lack of defined operators on some of the Bitcoind types made my life difficult.

**Review by EyeOfPython — APPROVED** [review]
Looks good now.

**Pull request merged**